### PR TITLE
Decompile functions in the func_8009ED70 call chain

### PR DESCRIPTION
**Pull request by joshlory**
```mermaid
stateDiagram-v2
    func_8009ED70:::g --> func_8002BB80:::b
    func_8009ED70:::g --> func_8002B160:::b
    func_8009ED70:::g --> func_8002B288:::g
    func_8009ED70:::g --> func_8002D9BC:::b
    func_8002D9BC:::b --> func_8002BB80:::b
    func_8009ED70:::g --> func_800AF808:::g
    func_800AF808:::g --> func_800AF828:::g
    func_800AF828:::g --> func_800AFAB4:::g
    func_800AFAB4:::g --> func_8002AD3C:::b
    func_800AF828:::g --> func_8002B73C:::g
    func_8009ED70:::g --> func_80015DC8:::r
    func_8009ED70:::g --> func_8002C614:::g
    func_8002C614:::g --> ...:::r
    func_8009ED70:::g --> func_8002B694:::g
    classDef r fill:#f00,color:white,stroke:#c00
    classDef g fill:#0c0,color:white,stroke:#0a0
```

### CODE REVIEW

File: src/main/323C.c [hunk 1]
```diff
@@ -1469,7 +1467,26 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B160);
...
+    arg0->unk3 = 0;
+    if (arg0->unk14 < 0) {
+        var_v0 = arg0->x_pos;
+        var_a0 = arg0->y_pos;
+    } else {
+        temp_v0 = arg0->unk14 * 0x54 / 2;
+        var_v0 = arg0->x_pos - D_801419BA[temp_v0];
+        var_a0 = arg0->y_pos - D_801419BE[temp_v0];
+    }
+    if ((((var_v0 + 0x20) & 0xFFFF) < 0x180U) && (((var_a0 + 0x20) & 0xFFFF) < 0x130U)) {
```
Comment (joshlory) [inline src/main/323C.c:1486]: There's probably a cleaner way to write this but I haven't found it yet.

File: src/main/323C.c [hunk 2]
```diff
@@ -1491,15 +1508,42 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B468);
...
+
+    if (arg0->unk15) {
+        arg0->unk20 += arg0->unk28;
+    } else {
+        arg0->unk20 -= arg0->unk28;
+    }
+
+    arg0->unk24 -= arg0->unk2C;
+    if (arg0->unk24 < FIXED(-6.5)) {
+        arg0->unk24 = FIXED(-6.5);
```
Comment (joshlory) [inline src/main/323C.c:1524]: Good sign that some of these fields are using fixed-point types.

Comment (sozud) [reply src/main/323C.c:1524]: Seems like we should go with x_pos, y_pos as the f32 type?
https://decomp.me/scratch/LCm6C
https://decomp.me/scratch/zemvr

File: src/main/323C.c (hunk 1 above)
Comment (sozud) [reply src/main/323C.c:1486]: Try this? https://decomp.me/scratch/zemvr

File: src/main/323C.c (hunk 2 above)
Comment (joshlory) [reply src/main/323C.c:1524]: Yup, makes sense! Do you want to do that as part of this PR or a follow-up? (feel free to push commits to this branch, or I can look in a bit)

Comment (sozud) [reply src/main/323C.c:1524]: It doesn't have to be done right now, the pattern was becoming clear so figured I'd mention it

File: src/main/323C.c (hunk 1 above)
Comment (joshlory) [reply src/main/323C.c]: Thanks!

File: src/main/323C.c (hunk 2 above)
Comment (joshlory) [reply src/main/323C.c:1523]: Easy enough to do that cleanup now. I also had hi and lo reversed, so this fixes that.

**Review by sozud — APPROVED** [review]

**Pull request merged**